src/sage/sets/recursively_enumerated_set.pyx

```diff
@@ -406,7 +386,7 @@ def RecursivelyEnumeratedSet(seeds, successors, structure=None,
 
     .. WARNING::
 
-        If you do not set the good structure, you might obtain bad results,
+        If you do not set a valid structure, you might obtain bad results,
```

Comment:
```suggestion
        If you do not set a good structure, you might obtain bad results,
```
Valid is not quite strong enough here I think. In either case, it is better to be a bit more vague.

Reply:
OK, I am surprised to learn that "good" is *not* frenglish!

More importantly, I don't understand the sentence, it seems contradictory to me:
* valid is *not* strong enough
* it is *better* to be more vague

Reply:
The term "good" is not well-defined (more vague), but it implies that there could be something more needed than what is specified as valid (stronger conditions are needed).

Reply:
Does the following sound good?
Instead of the statement:
`If you do not set a valid/ good structure, you might obtain bad results, like elements generated twice:`
            
we may write:

`A poorly defined recursion might lead to an unexpected result, such as some of the elements being generated multiple times:`

Reply:
No, that isn't quite right and becomes too vague as it doesn't reference the structure of the poset/recursion. It would be best just to leave it as "good" with changing "the" to "a".